Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs

```diff
@@ -160,14 +249,27 @@ public static void GracefulProcessShutdown ([NotNull] Process process, TimeSpan
         // https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill#System_Diagnostics_Process_Kill
         // This exception can be swallowed safely, as we are sure that a process is associated and the process has
         // already exited, therefore the method can return at this point.
-        return;
       }
+    }
 
-      // No exception is thrown when calling .WaitForExit(..) on an already exited process.
-      if (process.WaitForExit (timeoutInMilliseconds))
-        return;
+    private static void WaitForProcessesExits (List<Process> processes, TimeSpan timeout)
+    {
+      // Go through all processes and call .WaitForExit on each using a shared timeout
+      // If the timeout runs out still go through all remaining process to make sure to clear any exited processes
+      var timeoutStart = DateTime.UtcNow;
+      for (var i = 0; i < processes.Count; i++)
+      {
+        var elapsedTimeSinceTimeoutStart = DateTime.UtcNow - timeoutStart;
+        var timeToTimeoutEnd = timeout - elapsedTimeSinceTimeoutStart;
```

Comment:
Is there a benefit in using the following `DateTime` calc logic over just using a `Stopwatch`? I think it would make the logic a litte more concise and understandable.